config/routes.rb

```diff
@@ -13,7 +13,9 @@
   get 'reservations/confirm', to:'reservations#confirm', as:'confirm_reservation'  
   delete 'reservations/confirm/destroy/:id', to:'reservations#destroy', as:'destroy_reservation'
   
-  get 'users/show', to:'users#show', as:'show_user'
+  get '/signup', to:'users#new', as:'new_user'
+  post '/signup', to:'users#create', as:'create_user'
+  get 'users/show/:id', to:'users#show', as:'show_user'
```

Comment:
これだとRESTAPIになっていないのでこのようにするか、
```suggestion
  get 'users/:id', to:'users#show'
```
以下のようにするのがよいです。（onlyは必要によって変えてください）
```rb
resources :users, only: %i(show update)
```